packages/vite/src/client/client.ts

```diff
@@ -208,6 +208,10 @@ async function handleMessage(payload: HotPayload) {
             return hmrClient.queueUpdate(update)
           }
 
+          if (!hasDocument) {
+            return
+          }
+
```

Comment:
I think the only change needed is this line?

Reply:
Why would we only change this line? There are many DOM operations that will fail in workers, so it seemed best to make each one of them safe. Even if some may be safe today (e.g., first `hasDocument` check prevents others from running), if the code paths change at all they will throw errors as well.

Reply:
I agree that it is still somewhat error prone, and there are other latent bugs, such as `location.reload()`, which isn't available in workers either.

I suppose we could make the client more headless with hooks and guard all the "has DOM" things in one spot, but that would require some extra refactoring.